notifications/src/main/kotlin/com/amazon/opendistroforelasticsearch/commons/notifications/model/FeatureConfig.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+package com.amazon.opendistroforelasticsearch.commons.notifications.model
+
+import com.amazon.opendistroforelasticsearch.notifications.util.logger
+import com.amazon.opendistroforelasticsearch.notifications.util.valueOf
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.io.stream.Writeable
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+
+/**
+ * Data class representing Notification config for exposed for other plugins.
+ */
+data class FeatureConfig(
+    val name: String,
+    val description: String,
+    val configType: NotificationConfig.ConfigType,
+    val isEnabled: Boolean = true
+) : BaseModel {
```

Comment:
how do we handle that channels are avaliable to certain features? I didn't see any filtering/querying for that.

Reply:
the request has features. so the result is expected to contain only allowed ones for the feature.

Reply:
oh yes, I missed that.